### PR TITLE
Incoming messages from external connector

### PR DESCRIPTION
**Pull request by sergey-chernikov**

### CODE REVIEW

File: BlockSettleUILib/Trading/DealerXBTSettlementContainer.cpp [hunk 1]
```diff
@@ -286,7 +293,46 @@ void DealerXBTSettlementContainer::onTXSigned(unsigned int idReq, BinaryData sig
...
+         return;  // not our pay-out
+      }
+      const auto &serializedTx = tx.serialize();
+      const auto &buyAuthKey = weSellXbt_ ? reqAuthKey_ : authKey_;
+      const auto &sellAuthKey = weSellXbt_ ? authKey_ : reqAuthKey_;
+      const auto &result = bs::TradesVerification::verifySignedPayout(serializedTx
+         , buyAuthKey.toHexStr(), sellAuthKey.toHexStr(), expectedPayinHash_
+         , bs::XBTAmount(amount_).GetValue(), utxoReservationManager_->feeRatePb()
+         , settlementIdHex_, settlAddr_.display());
+      if (result->success) {
```
Comment (Ation) [inline BlockSettleUILib/Trading/DealerXBTSettlementContainer.cpp:320]: There is a very small chance that fee rate will be changed while we are waiting for TX on chain. I think that we could ignore it

File: BlockSettleUILib/UserScriptRunner.cpp [hunk 2]
```diff
@@ -363,6 +375,42 @@ void AQScriptHandler::settled(const std::string &quoteReqId)
    });
 }
 
+void AQScriptHandler::extMsgReceived(const std::string &data)
+{
+   QJsonParseError jsonError;
+   const auto &jsonDoc = QJsonDocument::fromJson(QByteArray::fromStdString(data)
+      , &jsonError);
+   if (jsonError.error != QJsonParseError::NoError) {
+      logger_->error("[AQScriptRunner::onExtDataReceived] invalid JSON message: {}"
```
Comment (Ation) [inline BlockSettleUILib/UserScriptRunner.cpp:384]: wrong function name in log text

Comment (sergey-chernikov) [reply BlockSettleUILib/UserScriptRunner.cpp]: Thank you, fixed

File: BlockSettleUILib/UserScriptRunner.cpp [hunk 3]
```diff
@@ -363,6 +375,42 @@ void AQScriptHandler::settled(const std::string &quoteReqId)
...
+   const auto &strMsg = QString::fromStdString(msgDoc.toJson(QJsonDocument::Compact).toStdString());
+   if (strFrom.isEmpty() || strType.isEmpty() || msgObj.isEmpty()) {
+      logger_->error("[AQScriptHandler::extMsgReceived] invalid data in JSON: {}"
+         , data);
+      return;
+   }
+
+   extDataPool_.push_back({strFrom, strType, strMsg});
+   while (extDataPool_.size() > maxExtDataPoolSize_) {
+      extDataPool_.pop_front();
```
[the remark this answers is not in the document]
Comment (sergey-chernikov) [reply BlockSettleUILib/UserScriptRunner.cpp:402]: This member can be accessed in Qt thread, but not in onMDUpdate - only after creating a new AQ script instance.

**Pull request merged**